gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -139,6 +188,28 @@ def run(
         return wti_list, ta_metrics
 
 
+def _sample_kde_directions(w_i2Ui1_measurements: BinaryMeasurementsUnit3, num_samples: int) -> List[Unit3]:
+    """Fits a Gaussian density kernel to the provided measurements, and then samples num_samples from this kernel.
+
+    Args:
+        w_i2Ui1_measurements: List of BinaryMeasurementUnit3 direction measurements.
+        num_samples: Number of samples to be sampled from the kernel.
+
+    Returns:
+        List of sampled Unit3 directions.
+     """
+    w_i2Ui1_list = [w_i2Ui1.measured() for w_i2Ui1 in w_i2Ui1_measurements]
+    if len(w_i2Ui1_list) > MAX_KDE_SAMPLES:
+        w_i2Ui1_list = [w_i2Ui1_list[i] for i in random.sample(range(len(w_i2Ui1_list)), MAX_KDE_SAMPLES)]
```

Comment:
maybe cleaner to only use random number generator from numpy instead of `random` and `numpy`.

could we use
```python
random.choice(a=w_i2Ui1_list, size=MAX_KDE_SAMPLES, replace=False)
```